lib/Sema/ConstraintSystem.cpp

```diff
@@ -6018,7 +6018,8 @@ Expr *ConstraintSystem::buildTypeErasedExpr(Expr *expr, DeclContext *dc,
 
   auto *decl = dyn_cast_or_null<ValueDecl>(dc->getAsDecl());
   if (!decl ||
-      !(decl->isDynamic() || decl->getDynamicallyReplacedDecl()))
+      (!Context.LangOpts.hasFeature(Feature::OpaqueTypeErasure) &&
+       !(decl->isDynamic() || decl->getDynamicallyReplacedDecl())))
```

Comment:
Could you please update the doc comment for `buildTypeErasedExpr` in ConstraintSystem.h to specify that either the function is dynamically replaceable or experimental opaque type erasure is enabled?